PlaceAnnotationView 적용

randychoi95 opened the pull request:
- PlaceAnnotationView를 재사용하기 위해, mapview에 재사용할 PlaceAnnotationView 등록
- MKMapviewDelegate를 채택하여 PlaceAnnotationView을 사용하게 구현, PlaceAnnotationView 등록시 애니메이션 적용, PlaceAnnotationView 선택시 위치 이동 적용

<img src = "https://github.com/f-lab-edu/PlaceStory/assets/60861313/3fae7f41-7ec8-4b61-a76a-166725a20aca" width="300">

Commit:
:recycle: PlaceAnnotationView 적용

- PlaceAnnotationView를 재사용하기 위해, mapview에 재사용할 PlaceAnnotationView 등록
- MKMapviewDelegate를 채택하여 PlaceAnnotationView을 사용하게 구현, PlaceAnnotationView 등록시 애니메이션 적용, PlaceAnnotationView 선택시 위치 이동 적용

f-lab-max commented on PlaceStory/MyLocation/Sources/MyLocation/MyLocationViewController.swift:
```diff
@@ -73,6 +75,46 @@ final class MyLocationViewController: UIViewController, MyLocationPresentable, M
         listener?.didTappedPlaceSearchButton()
     }
     
+    private func isDuplicateAnnotation(_ annotation: [MKAnnotation], _ currentLocation: CLLocationCoordinate2D) -> Bool {
```
뷰에는 로직이 가급적 로직이 없어야 합니다. 뷰는 최대한 수동적으로 그리라는대로 그리고, 이벤트는 그대로 전달만 하는 식으로 짜줘야합니다.

f-lab-max commented on PlaceStory/MyLocation/Sources/MyLocation/MyLocationViewController.swift:
```diff
@@ -73,6 +75,46 @@ final class MyLocationViewController: UIViewController, MyLocationPresentable, M
+        self.myPlaceMapView.mapView.addAnnotation(placeAnnotation)
+    }
+    
+    private func configureAnnotationView(for annotation: PlaceAnnotation, on mapView: MKMapView) -> MKAnnotationView {
```
아직도 뷰에 MapKit 의존성이 남아있네요. 이 의존성도 없애주셔야 합니다

Commit:
:refactor: 리뷰 반영

- annotation 중복 확인 로직을 interactor로 옮기고, 중복이 안되는 annotation을 추가하도록 view에 요청하는 구조로 변경
- configureAnnotationView 메소드 제거

Commit:
:refactor: 리뷰 반영

- viewController와 mapkit 분리
- MKMapViewDelegate를 대신 처리하는 객체 구현
- ProxyPackage에서 CommonUI로 MapView 구현

f-lab-max reviewed (approved):
일단 approve를 하고, 멘토링 연장 세션에서 라이브로 같이 수정해보면 좋을거 같아요. 수고하셨습니다!
